File: CPP/11. Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;

        int res = INT_MIN;
        while(left < right){
            res = max(res, (right-left) * min(height[left], height[right]) );
            if(height[left] <= height[right]){
                left++;
            }
            else{
                right--;
            }
        }

        return res;
    }
};

File: CSharp/11. Container With Most Water.cs
public class Solution {
    public int MaxArea(int[] height) {
        int left = 0;
        int right = height.Length- 1;
        int area = Int32.MinValue;
        while(left < right){
            area = Math.Max(area, (right-left)*Math.Min(height[left], height[right]));
            if(height[left] < height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        
        return area;
    }
}